src/legacy/ui/public/agg_types/controls/drop_partials.tsx

```diff
@@ -0,0 +1,58 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import React from 'react';
+
+import { EuiFormRow, EuiIconTip, EuiSwitch } from '@elastic/eui';
+import { i18n } from '@kbn/i18n';
+import { FormattedMessage } from '@kbn/i18n/react';
+import { AggParamEditorProps } from '../../vis/editors/default';
+
+function DropPartialsParamEditor({ agg, aggParam, value, setValue }: AggParamEditorProps<boolean>) {
+  if (agg.params.field.name !== agg.getIndexPattern().timeFieldName) {
+    return null;
+  }
+
+  const content = i18n.translate('common.ui.aggTypes.dropPartialBucketsTooltip', {
+    defaultMessage:
+      "Remove buckets that span time outside the time range so the histogram doesn't start and end with incomplete buckets.",
+  });
+
+  const label = (
+    <>
+      <FormattedMessage
+        id="common.ui.aggTypes.dropPartialBucketsLabel"
+        defaultMessage="Drop partial buckets"
+      />{' '}
+      <EuiIconTip position="right" content={content} type="questionInCircle" aria-label={content} />
+    </>
+  );
+
+  return (
+    <EuiFormRow label={label} fullWidth={true} className="visEditorSidebar__aggParamFormRow">
+      <EuiSwitch
+        checked={Boolean(value)}
```

Comment:
Why do we need the `Boolean` here? `value` should always be of type boolean (or otherwise our typing of `AggParamEditorProps` is false :-)

Reply:
When the component renders for the first time the `value` is `undefined`

![image](https://user-images.githubusercontent.com/31325372/55473550-a856ad00-5617-11e9-9f82-a08496da8c90.png)

and ends up with react warning

![image](https://user-images.githubusercontent.com/31325372/55473634-ce7c4d00-5617-11e9-9d0c-c9203000a770.png)

Reply:
cc @maryia-lapata then our typing of `AggParamEditorProps` is wrong and it should rather be `value?: T`, if every value can apparently also be `undefined`. Of course it would be nicer if it wouldn't be undefined, but maybe it's easier for now, just allow it to be undefined instead of making sure in some central part that it always has a default value (especially since that central part isn't using TypeScript yet).

Reply:
@timroes I debugged and noticed that during the first run `value` is `undefined` in `AggParamReactWrapper` though `paramValue` was set in watcher before. I assume that's because that  React component is transformed into angular directive.
But if we initialize `paramValue` outside the watcher, `AggParamReactWrapper` will be invoked with defined `value`.

Reply:
Perfect finding. @sulemanof can you try adding in https://github.com/elastic/kibana/blob/master/src/legacy/ui/public/vis/editors/default/agg_param.js#L90 after line 90 a:

```
$scope.paramValue = $scope.agg.params[$scope.aggParam.name];
```

That will hopefully help that this is not `undefined`.

Reply:
@timroes Yep, that helped. Thanks =)
PR updated